include/systems/implicit_system.h

```diff
@@ -34,16 +34,26 @@ template <typename T> class LinearSolver;
 template <typename T> class SparseMatrix;
 
 /**
- * This class provides a specific system class.  It aims
- * at implicit systems, offering nothing more than just
- * the essentials needed to solve a system.
+ * \brief Manages consistently variables, degrees of freedom, coefficient
+ * vectors, and matrices for implicit systems.
+ *
+ * This class provides a specific system class. It aims at implicit systems.
```

Comment:
These two sentences from the original docs are not very descriptive since no terms are actually defined, I think we could probably drop them.